ui/components/ui/token-input/token-input.component.js

```diff
@@ -127,7 +129,7 @@ export default class TokenInput extends PureComponent {
     } else {
       // Display ETH
       currency = ETH;
-      numberOfDecimals = 6;
+      numberOfDecimals = primaryNumberOfDecimals;
```

Comment:
Use of decimal from token details instead of the hardcoding to 6 seems to be a good change to add even if we decide against decimal value as 18